### PR TITLE
[Silabs] [EFR32] Refactor 917 NCP and 917 SoC Bluetooth "DriveBLEState" function and bugfix RS9116 Stop Advertising

### PR DESCRIPTION
**Pull request by shgutte**
Problem
There is DriverBLEState function code flow is not working properly for RS9117 and RS9116

Fixes added
Fixed the code flow of the DriverBLEState function and fix the Stop Advertisement issue

Testing
I tested with all combinations of RS9116 and RS9117.

### CODE REVIEW

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 1]
```diff
@@ -506,23 +496,21 @@ CHIP_ERROR BLEManagerImpl::MapBLEError(int bleErr)
 
 void BLEManagerImpl::DriveBLEState(void)
 {
-    ChipLogProgress(DeviceLayer, "DriveBLEState starting");
+    SILABS_LOG("%s starting", __func__);
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:499]: I recommend stepping away from using SILABS_LOG except if it is something on the app side.

Also, this print doesn't provide much except specific debugging of the function flow. I suggest removing this trace for all together.

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 2]
```diff
@@ -506,23 +496,21 @@ CHIP_ERROR BLEManagerImpl::MapBLEError(int bleErr)
...
     if (mServiceMode == ConnectivityManager::kCHIPoBLEServiceMode_Enabled && mFlags.Has(Flags::kAdvertisingEnabled) &&
         NumConnections() < kMaxConnections)
     {
-
-        ChipLogProgress(DeviceLayer, "Start/re-start advertising if not already started, or if there is a pending change");
         // Start/re-start advertising if not already started, or if there is a pending change
         // to the advertising configuration.
         if (!mFlags.Has(Flags::kAdvertising) || mFlags.Has(Flags::kRestartAdvertising))
         {
+            SILABS_LOG("Start Advertising");
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:513]: use ChipLogProgress

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 3]
```diff
@@ -531,12 +519,13 @@ void BLEManagerImpl::DriveBLEState(void)
     // Otherwise, stop advertising if it is enabled.
     else if (mFlags.Has(Flags::kAdvertising))
     {
+        SILABS_LOG("Stop Advertising");
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:522]: use ChipLogProgress

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 4]
```diff
@@ -531,12 +519,13 @@ void BLEManagerImpl::DriveBLEState(void)
...
         err = StopAdvertising();
         SuccessOrExit(err);
     }
 
 exit:
-    ChipLogProgress(DeviceLayer, "DriveBLEState End");
+    SILABS_LOG("%s End", __func__);
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:528]: excessive logging. It is only helpful for active debugging. I suggest removing it.

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 5]
```diff
@@ -718,19 +711,16 @@ void BLEManagerImpl::UpdateMtu(rsi_ble_event_mtu_t evt)
 
 void BLEManagerImpl::HandleBootEvent(void)
 {
+    ChipLogProgress(DeviceLayer, "HandleBootEvent started");
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:714]: excessive logging.

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp [hunk 6]
```diff
@@ -97,7 +97,7 @@ void sl_ble_init()
     rsi_ble_app_init_events();
 
     SILABS_LOG("StartAdvertising");
```
Comment (jmartinez-silabs) [inline src/platform/silabs/rs911x/BLEManagerImpl.cpp:99]: You have a print for start Advertising here and also in the drive ble state

I'd remove this one. 

Also use the chip logging API not SILABS_LOG, even more so for non app specific logging

**Review by jmartinez-silabs — CHANGES_REQUESTED** [review]

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 1 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp:499]: Removed the log

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 2 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: Updated

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 3 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: Updated

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 4 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: Removed

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 5 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: Removed

File: src/platform/silabs/rs911x/BLEManagerImpl.cpp (hunk 6 above)
Comment (shgutte) [reply src/platform/silabs/rs911x/BLEManagerImpl.cpp]: Removed

**Review by jmartinez-silabs — APPROVED** [review]

**Review by andy31415 — APPROVED** [review]

**Pull request merged**